src/pycram/process_modules/icub_process_modules.py

```diff
@@ -0,0 +1,687 @@
+import math
+from threading import Lock
+from typing import Union
+from typing_extensions import Any
+import actionlib
+from .default_process_modules import DefaultMoveJoints, DefaultMoveArmJoints, DefaultMoveTCP, DefaultNavigation, \
+    DefaultMoveHead, DefaultDetecting, DefaultMoveGripper
+from .. import world_reasoning as btr
+import numpy as np
+from ..process_module import ProcessModule, ProcessModuleManager
+from ..external_interfaces.ik import request_ik
+from ..ros.logging import logdebug
+from ..utils import _apply_ik
+from ..local_transformer import LocalTransformer
+from ..designators.object_designator import ObjectDesignatorDescription
+from ..designators.motion_designator import MoveMotion, LookingMotion, \
+    DetectingMotion, MoveTCPMotion, MoveArmJointsMotion, WorldStateDetectingMotion, MoveJointsMotion, \
+    MoveGripperMotion, OpeningMotion, ClosingMotion
+from ..robot_description import RobotDescription
+from ..datastructures.world import World
+from ..world_concepts.world_object import Object
+from ..datastructures.pose import Pose
+from ..datastructures.enums import JointType, ObjectType, Arms, ExecutionType, GripperState
+from ..external_interfaces import giskard
+from ..external_interfaces.robokudo import *
+import yarp
```

Comment:
This add yarp as a hard dependency. Wrap this into a try, catch and add a log function in case the imports fail. Look [here](https://github.com/cram2/pycram/blob/dev/src/pycram/external_interfaces/giskard.py#L22) for reference

Reply:
Done